### ISSUE TITLE
Discuss and enforce a packaging policy

### ISSUE THREAD

**Issue by livingsilver94** [serpent-os/recipes#148]
We have the luxury of a new beginning, we can decide and enforce a packaging policy we all agree upon.

The following list in the top-level comment is the current proposed policy. An unchecked item is an item we still have to discuss and agree upon. A checked item is agreed upon.

#### Naming
- [ ] Package names must contain only characters in this ASCII subset: `a` to `z`, `0` to `9`, `-`, `_`.
- [ ] Dash (`-`) must be used either as a separator, or within the upstream name of the package. For example in `glibc-devel` the dash is the separator between the name and the development subpackage; in `super-useful-app` the dashes are part of the upstream name "Super-useful-app".
- [ ] Underscore (`_`) must only be used as part of the upstream name, never as a separator.
- [ ] Packages that are part of a family must be prefixed with the name, or abbreviation, of that family. For example, Python packages may have the `python-` prefix, KDE Framework based on Qt6 may have the `kf6-` prefix, etc.
- [ ] If a package provides development resources but it's meant to be primarily a package for regular users, it must have a development subpackage named after the parent package, plus the `-devel` suffix, containing the development resources. If a package provides **only** development resources, there won't be any `-devel` subpackage since there are no resources to split.

#### Dependencies
 - [ ] `-devel` subpackages must depend on their parent pacakge.
 - [ ] `-devel` subpackages must not be a dependency of non-`-devel` packages, unless such packages are **only** for development. For example, installing `-devel` packages along with `gcc` is fine, since `gcc` is a compiler.

**Comment by livingsilver94** [comment]
We may also want to discuss the usage of "must" vs "should".

**Comment by joebonrichie** [comment]
i vote to get rid of underscores entirely

**Comment by Staudey** [comment]
Also worth thinking about naming convention for ABI-compatibility packages like e.g. `glibmm-2.68` or `libwebkit-gtk5`

**Comment by livingsilver94** [comment]
Do you have rights to edit the top-level comment? If so, you may propose your solution.

**Comment by Staudey** [comment]
No, I don't have edit permissions (I've never been one of you snake people ^^). Anyway, I have no thought-out solution (yet); just wanted to make a mention of the topic for now.

**Comment by livingsilver94** [comment]
Proposed convention added! Plus, I included the dot among the allowed characters.

**Comment by human222** [comment]
I propose:

1. Purge all the .files and .folders from the home directory. They are a massive source of clutter. Put the user's configuration files/folders inside a "config" subdirectory instead.
2. Include the package's version number (or at least the major version number) in the name. python3 may be the "current" thing, but it should still be named python3, not python.
3. Keep multiple versions of packages unless compatibility is certain. Languages and game engines often break compatibility between versions.

**Comment by NomadicCore** [comment]
@CookieSource 

Just doing some housekeeping and came by this discussion. We should consider a page for this on our documentation site as I don't think we go into anywhere near enough detail on naming convention.

**Comment by CookieSource** [comment]
> [@CookieSource](https://github.com/CookieSource)
> 
> Just doing some housekeeping and came by this discussion. We should consider a page for this on our documentation site as I don't think we go into anywhere near enough detail on naming convention.

I think it's a good idea once we make a proper policy until then I think it wouldn't be a great idea?
